src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOp.java

```diff
@@ -121,11 +121,17 @@ Op innerMostEnclosingTarget() {
                 if (op == null) {
                     throw new IllegalStateException("No enclosing loop");
                 }
-            } while (!(op instanceof Op.Loop));
+            } while (!(op instanceof Op.Loop || op instanceof JavaSwitchStatementOp));
             // } while (!(op instanceof Op.Loop lop));
             // error: variable lop might not have been initialized
```

Comment:
Now we have two choices we can remove the comment. Also we can update the comment at line 114.

Suggest you also change the if/else if/else into a switch expression.